delta/src/main/scala/io/qbeast/spark/delta/DeltaQbeastSnapshot.scala

```diff
@@ -16,26 +16,31 @@
 package io.qbeast.spark.delta
 
 import io.qbeast.core.model._
+import io.qbeast.spark.index.IndexStatusBuilder
 import io.qbeast.spark.utils.MetadataConfig
 import io.qbeast.spark.utils.TagColumns
 import io.qbeast.IISeq
 import org.apache.hadoop.fs.Path
 import org.apache.spark.sql.delta.actions.AddFile
+import org.apache.spark.sql.delta.DeltaLog
 import org.apache.spark.sql.delta.Snapshot
 import org.apache.spark.sql.functions.lit
+import org.apache.spark.sql.types.StructType
 import org.apache.spark.sql.AnalysisExceptionFactory
 import org.apache.spark.sql.DataFrame
 import org.apache.spark.sql.Dataset
+import org.apache.spark.sql.SparkSession
 
 /**
  * Qbeast Snapshot that provides information about the current index state.
  *
- * @param snapshot
- *   the internal Delta Lakes log snapshot
+ * @param tableID
+ *   the table ID
  */
-case class DeltaQbeastSnapshot(protected override val snapshot: Snapshot)
-    extends QbeastSnapshot
-    with DeltaStagingUtils {
+case class DeltaQbeastSnapshot(tableID: QTableID) extends QbeastSnapshot with DeltaStagingUtils {
```

Comment:
Having the `Snapshot` attached to the `TableId` would give you access all the time to the most updated snapshot, which conflicts with the terminology. The concept of Snapshot should be an isolated abstraction of the state of a table for a given moment. 

The most immediate consequence is that removes the possibility of adding time travel (see issue #429 ). 


Again, not for this PR. But agree to open an issue and document it, if we agree. Possibily it's not necessary to build it asap.

Reply:
opened issue #448